### PR TITLE
Camera

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
Addresses https://github.com/smashingboxes/fraqture/issues/49

### What Changed?
Added live camera integration via `imagesnap`
Snapshots are stored in `/images` and automatically enter the cycle

### CODE REVIEW

File: src/fraqture/photo_countdown.clj [hunk 1]
```diff
@@ -14,23 +15,32 @@
...
-      (q/text (str (:time-left state)) center-width center-height)
+      (q/text (str (:time-left state)) center-width center-height))
+    (when (< (:time-left state) 1)
+      (q/text "" center-width center-height)
       (led/paint-window serial 0 0 led/row-count led/col-count [255 255 255]))
-    (q/delay-frame 1000)))
+    (led/refresh serial)
+
+    (if (= (:time-left state) -1)
+      (shell/sh "imagesnap" "-w" "1 " (str "images/" (now) ".jpg")))
```
Comment (sweetmandm) [inline src/fraqture/photo_countdown.clj]: I had trouble identifying how the `imagesnap` dependency is managed -- is the intention that I install it manually, or am I missing something?

Comment (TylerRockwell) [inline src/fraqture/photo_countdown.clj]: It currently needs to be manually installed. You can just `brew install imagesnap`. I'll add this to the README, though for the sake of clarity.

File: src/fraqture/photo_countdown.clj [hunk 2]
```diff
@@ -14,23 +15,32 @@
...
         center-width (/ (q/width) 2)
         center-height (/ (q/height) 2)]
     (if (> (:time-left state) 0)
-      (q/text (str (:time-left state)) center-width center-height)
+      (q/text (str (:time-left state)) center-width center-height))
+    (when (< (:time-left state) 1)
+      (q/text "" center-width center-height)
       (led/paint-window serial 0 0 led/row-count led/col-count [255 255 255]))
-    (q/delay-frame 1000)))
+    (led/refresh serial)
```
Comment (sweetmandm) [inline src/fraqture/photo_countdown.clj]: Is it necessary to `led/refresh` on each draw, or would it only really be necessary inside the `when` above? Not that it should be changed as long as it's safe and low-cost, I'm just trying to figure out what it's doing until that `when` evaluates true.

Comment (TylerRockwell) [inline src/fraqture/photo_countdown.clj]: It can probably be added to the `when` above. It doesn't do anything with the leds before that.

**Comment by sweetmandm** [comment]
looks good to me! 👍

**Comment by TylerRockwell** [comment]
Thanks, @sweetmandm !

**Pull request merged**